flycheck.el

```diff
@@ -6381,7 +6378,8 @@ Requires GCC 4.8 or newer.  See URL `https://gcc.gnu.org/'."
    (info line-start (or "<stdin>" (file-name)) ":" line ":" column
          ": note: " (message) line-end)
    (warning line-start (or "<stdin>" (file-name)) ":" line ":" column
-            ": warning: " (message) line-end)
+            ": warning: " (message (one-or-more (not (any "\n["))))
+            (optional "[" (id (one-or-more (any "A-Z-"))) "]") line-end)
```

Comment:
Ha!  That did not even phase me.  But on a second reading, I'm puzzled.

I've looked into it, and we don't seem to pass anything related to case-sensitivity in Flycheck.  So I guess case-sensitivity is controlled by `case-fold-search`.

To be safe, I've changed the regexp to what other checkers do: `(one-or-more not-newline)`.